### PR TITLE
ft(S3C-3020): Add functional tests for IngestShards

### PR DESCRIPTION
**Pull request by tmacro**

### CODE REVIEW

File: tests/functional/v2/task/testIngestShards.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+const getClient = prefix => new CacheClient({
+    backend: new cacheBackends.RedisCache(
+        config.redis,
+        prefix,
+    ),
+});
+
+function eventToWarp10(event) {
+    // console.log(Object.entries(event.getValue()).filter(([key]) => eventFieldsToWarp10[key]))
```
Comment (jonathan-gramain) [inline tests/functional/v2/task/testIngestShards.js]: remove this comment?

File: tests/functional/v2/task/testIngestShards.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    return Object.entries(event.getValue())
+        .filter(([key]) => eventFieldsToWarp10[key])
+        .reduce((e, [k, v]) => {
+            e[eventFieldsToWarp10[k]] = v;
+            return e;
+        }, {});
+}
+
+function assertResults(events, results) {
+    assert(results.result.length === 1);
```
Comment (jonathan-gramain) [inline tests/functional/v2/task/testIngestShards.js]: could be better written as `assert.strictEqual(results.result.length, 1)`, same for other similar asserts

**Review by jonathan-gramain — APPROVED** [review]

**Review by naren-rajendran — APPROVED** [review]

**Pull request merged**